app/src/main/java/org/sopt/and/MyActivity.kt

```diff
@@ -0,0 +1,123 @@
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.widthIn
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.sopt.and.User.EMAIL
+import org.sopt.and.component.MyPageBox
+import org.sopt.and.component.MyPagePurchaseBox
+import org.sopt.and.ui.theme.ANDANDROIDTheme
+
+class MyActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val intent = intent
+        val email = intent.getStringExtra(EMAIL) ?: ""
+        setContent {
+            ANDANDROIDTheme {
+                Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
+                    MyPageScreen(
+                        modifier = Modifier.padding(innerPadding),
+                        email = email
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun MyPageScreen(modifier: Modifier, email: String) {
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(color = Color(0xFF1B1B1B)),
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(color = Color(0xFF252525))
+                .padding(top = 20.dp, start = 20.dp, end = 20.dp, bottom = 5.dp),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Icon(
+                painterResource(id = R.drawable.ic_my_page_person_24),
+                contentDescription = null,
+                tint = Color.White,
+                modifier = Modifier
+                    .padding(end = 10.dp)
+                    .size(60.dp)
+            )
+            BoxWithConstraints {
+                val maxWidth = maxWidth
+                Text(
+                    text = email,
+                    color = Color.White,
+                    fontSize = 20.sp,
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis,
+                    modifier = Modifier.widthIn(max = maxWidth * 0.7f)
```

Comment:
이 기능을 사용해서 공간을 할당하신 이유가 있으신가요??

Reply:
이메일 형식 특성상 길이가 길어지면 마이페이지 뷰에서 오른쪽 아이콘들이 보이지 않는 경우가 발생해서 overflow로 처리하였는데
그렇다고 너무 짧은 길이로 제한하면 볼수있는 정도가 너무 적어서 개인적인 생각으로 아이콘이 보이고 닉네임도 어느정도 보이는 0.7f정도로 할당하였습니다!

Reply:
현재 Row의 구조상 아래처럼 되어있을 것 같아요. 📂는 icon, email뒤 여백은 Spacer(1f)를 의미해요
`📂 parkdongmin123@...  📂 📂`

물론 바형 스마트폰만 생각한다면 크게 달라지지는 않겠지만, 갤럭시 폴드의 예시를 들어볼게요

Icon 3개는 전부 크기가 dp로 할당되어있기 때문에 기기에 따라 커지긴 하지만 극적인 대응은 힘들어요
화면 넓이의 70%라는 값을 할당받은 Text는 동적으로 70%의 길이를 가져가겠지만, icon들이 그만큼 동적이지 못해 Spacer의 길이가 길어질거에요. 즉, 아래처럼 될 수도 있습니다.
`📂 parkdongmin123@gmail.c...               📂 📂`

이처럼 의도치 않은 Spacer의 크기가 할당 될 수 있어요.
그렇기 때문에 제가 제안하고 싶은 방법은 Spacer를 없애고, Text의 weight를 1f로 설정하는 거에요.

weight의 경우 다른 컴포넌트의 크기를 먼저 넣고, 남는 공간을 계산하여 최대 넓이를 정하는 방법이기 때문에 이 방법을 사용하신다면 Text가 길어지며 아이콘이 화면에 보이지 않는 현상이 없어질거에요 :)

```suggestion
                    modifier = Modifier.weight(1f)
```